src/Data/Docker/Nix/FetchDocker.hs

```diff
@@ -134,19 +150,23 @@ generateFetchDockerExpr dim@HockerImageMeta{..} configDigest layerDigests = do
         , StaticKey "imageName"
         ]
   let genLayerId i = mkSym . Text.pack $ "layer" <> show i
-  let fetchconfig = mkFetchDockerConfig (inherit $ ((StaticKey "tag"):commonInherits)) configDigest
+  let fetchconfig = mkFetchDockerConfig (inheritAdapter ("generated for " ++ show imageName) (mkPos 1) (mkPos 1) $ ((StaticKey "tag"):commonInherits)) configDigest
       fetchlayers =
         mkLets
-         (mkFetchDockerLayers (inherit commonInherits) layerDigests)
+         (mkFetchDockerLayers (inheritAdapter "common inherits" (mkPos 1) (mkPos 1) commonInherits) layerDigests)
          (mkList $ fmap genLayerId [0..(Prelude.length layerDigests)-1])
   fetchDockerExpr <- mkFetchDocker dim fetchconfig fetchlayers
   pure
     (mkFunction
       (mkParamset
-        [ ("fetchdocker",       Nothing)
-        , ("fetchDockerConfig", Nothing)
+        [ ("fetchDockerConfig", Nothing)
         , ("fetchDockerLayer",  Nothing)
-        ]) fetchDockerExpr)
+        , ("fetchdocker",       Nothing)
```

Comment:
FYI, this reordering was necessary because old hnix sorts them but new hnix preserves this order.